doc/sfp-cmis/cmis-init.md

```diff
@@ -0,0 +1,447 @@
+  * [About This Manual](#about-this-manual)
+  * [Scope](#scope)
+  * [Abbreviation](#abbreviation)
+  * [References](#references)
+  * [Requirement Overview](#requirement-overview)
+  * [Functional Description](#functional-description)
+  * [Design](#design)
+
+# List of Tables
+  * [Table 1: Definitions](#table-1-definitions)
+  * [Table 2: References](#table-2-references)
+
+# Revision
+| Rev |     Date    |       Author        | Change Description                        |
+|:---:|:-----------:|:-------------------:|-------------------------------------------|
+| 0.1 | 11/16/2021  | Dante (Kuo-Jung) Su | Initial version                           |
+
+# About this Manual
+This document provides general information about the CMIS application initialization
+support for SONiC.
+
+# Abbreviation
+
+# Table 1: Definitions
+| **Term**       | **Definition**                                   |
+| -------------- | ------------------------------------------------ |
+| pmon           | Platform Monitoring Service                      |
+| xcvr           | Transceiver                                      |
+| xcvrd          | Transceiver Daemon                               |
+| CMIS           | Common Management Interface Specification        |
+
+# References
+
+# Table 2 References
+
+| **Document**                                            | **Location**  |
+|---------------------------------------------------------|---------------|
+| CMIS v5 | [CMIS5p0.pdf](http://www.qsfp-dd.com/wp-content/uploads/2021/05/CMIS5p0.pdf) |
+
+# Requirement Overview
+
+This document describes functional behavior of the CMIS application initialization
+support in SONiC.
+
+The Common Management Interface Specification (CMIS) provides a variety of features
+and support for different transceiver form factors. A CMIS transceiver may support
+multiple application, and the application initialization sequence is now mandatory
+upon port mode changes. Otherwise the link will be down if the host port mode
+does not match the selected application on the CMIS transceiver.
+
+The feature is built on top of SONiC **sfp-refactor** framework to provide a
+platform-independent solution, and the individual platforms could easily enable
+this feaure by having its **Sfp** object inherited from **SfpOptoeBase**.
+
+**Example:**  
+```
+from sonic_platform_base.sonic_xcvr.sfp_optoe_base import SfpOptoeBase
+
+class Sfp(SfpOptoeBase):
+
+    def __init__(self, sfp_index):
+        SfpOptoeBase.__init__(self)
+        self.index = sfp_index
+
+    def get_port_type(self):
+        return self.SFP_PORT_TYPE_QSFPDD
+
+    def get_eeprom_path(self):
+        # platform-specific per-port I2C bus
+        bus_id = 32 + self.index
+        return "/sys/bus/i2c/devices/{}-0050/eeprom".format(bus_id)
+```
+
+This feature could also be disabled by the platform-specific **pmon_daemon_control.json**
+
+**Example:**  
+```
+$ cat sonic-buildimage/device/VENDOR/PLATFORM/pmon_daemon_control.json
```

Comment:
i don't see any reason why should some platform skip initializing QSFP-DD as per CMIS spec. As per the spec, we should minimize I2C transaction during datapath init:-
![image](https://user-images.githubusercontent.com/45705344/143797572-435793ae-d66c-45e0-a0ce-5b3d9a87931d.png)

Since Xcvrd is doing many i2c transactions for DOM and PM, it requires unnecessary synchronization b/w Xcvrd and proprietary firmware that is doing the DP init. @keboliu we can revisit this later when we understand why you want to skip initialization for your platform.

Reply:
As of now, we could add a sanity check to the DOM handelr of CMIS to skip this operation is the module state != READY